### DISCUSSION TITLE
Branch Workflow Update

### DISCUSSION THREAD

**Discussion by Khalid1G** [ISIL-ESTE/Student-Workflow-Organizer#31] [Announcements]
Hello Team 👋,

I would like to inform you about an important update regarding our branch workflow on GitHub. In order to streamline our development process and ensure a stable and reliable codebase, we have introduced a new branch called dev and implemented some changes to how we manage our main branch. Here are the details:

1. Creation and Protection of the dev Branch:
    -  I have created a new branch called `dev` which will serve as the primary branch for ongoing development.
    -  The `dev` branch is now protected, meaning that direct pushes or merges to this branch are restricted.
    
2. Purpose of the main Branch:
    - The `main` branch will now be reserved exclusively for tested and production-ready code.
    - We will deploy our project based on the code from the `main` branch.
    - This ensures that any code pushed to the main branch has undergone thorough testing and meets our quality standards.

3. Linking the Main Branch with the Server:
    - Going forward, the main branch will be directly linked with our deployment server.
    - This means that all new commits that pass the necessary testing on the `dev` branch will be automatically pushed to the main branch, triggering the deployment process.

By implementing this branch workflow update, we aim to achieve better code organization, enhanced stability, and efficient collaboration among the team. It also ensures that we maintain a clear distinction between development and production code.

> Please make sure to familiarize yourself with this new workflow and adjust your development processes accordingly. Should you have any questions or concerns, don't hesitate to reach out to me or any member of the team.

Thank you for your cooperation and dedication to delivering high-quality code.

**Comment by bellaabdelouahab** [comment]
@ISIL-ESTE
